librz/io/io.c

```diff
@@ -422,27 +422,26 @@ RZ_API char *rz_io_system(RzIO *io, const char *cmd) {
 	return NULL;
 }
 
-RZ_API bool rz_io_resize(RzIO *io, ut64 newsize) {
-	if (io) {
-		RzList *maps = rz_io_map_get_for_fd(io, io->desc->fd);
-		RzIOMap *current_map;
-		RzListIter *iter;
-		ut64 fd_size = rz_io_fd_size(io, io->desc->fd);
-		bool ret = rz_io_desc_resize(io->desc, newsize);
-		if (!ret) {
-			rz_list_free(maps);
-			return false;
-		}
-		rz_list_foreach (maps, iter, current_map) {
-			// we just resize map of the same size of its fd
-			if (current_map->itv.size == fd_size) {
-				rz_io_map_resize(io, current_map->id, newsize);
-			}
-		}
+RZ_API bool rz_io_resize(RZ_NONNULL RzIO *io, ut64 newsize) {
```

Comment:
Note that in that case all callers should check for NULL before this function called. Since those asserts are omitted from the release builds. If the bug did happen in the first place, that means there's a call passing NULL somewhere. Apart from that LGTM.

Reply:
In release build we have `checks_level = 1`:
https://github.com/rizinorg/rizin/blob/4594027f8b5ad7828941e59dce9fb023220bbabe/meson.build#L315-L319

In that case `rz_return_val_if_fail(expr, val)` works as a simple `if (!expr) return val;` 
https://github.com/rizinorg/rizin/blob/4594027f8b5ad7828941e59dce9fb023220bbabe/librz/include/rz_util/rz_assert.h#L108-L114

Reply:
Nice, didn't realize that. LGTM then.

Reply:
And the bug is caused by accessing `io->desc` which is `NULL`